nvflare/app_common/abstract/fl_model.py

```diff
@@ -11,84 +11,97 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
 from enum import Enum
 from typing import Dict, Optional
 
 
-class TransferType(Enum):
+class ModelType(str, Enum):
     MODEL = "MODEL"
     MODEL_DIFF = "MODEL_DIFF"
-    METRICS_ONLY = "METRICS_ONLY"
+    METRICS = "METRICS"
+    WEIGHTS = "WEIGHTS"
+    WEIGHT_DIFF = "WEIGHT_DIFF"
 
 
 class FLModelConst:
-    AGGREGATION = "aggregation"
+    MODEL_TYPE = "model_type"
+    MODEL = "model"
+    OPTIMIZER = "optimizer"
     METRICS = "metrics"
+    CONFIGS = "configs"
+    CLIENT_WEIGHTS = "client_weights"
+    ROUND = "round"
+    TOTAL_ROUNDS = "total_rounds"
+    META = "meta"
+    AGGREGATION = "aggregation"
 
 
 class FLModel:
     def __init__(
         self,
-        transfer_type: TransferType,
+        model_type: Optional[ModelType] = None,
         model: Optional[Dict] = None,
         optimizer: Optional[Dict] = None,
         metrics: Optional[Dict] = None,
         configs: Optional[Dict] = None,
         client_weights: Optional[Dict] = None,
         round: Optional[int] = None,
```

Comment:
@chesterxgchen should we call this `current_round` to avoid the name conflict with built-in round?

Reply:
Yes, we should.

Reply:
Let's change to current_round.

Reply:
current_round is also clearer I feel